pkg/odo/cli/component/create.go

```diff
@@ -370,10 +366,17 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 	}
 	co.DevfilePath = DevfilePath
 
-	if util.CheckPathExists(ConfigFilePath) || util.CheckPathExists(EnvFilePath) {
+	if util.CheckPathExists(ConfigFilePath) {
 		return errors.New("this directory already contains a component")
 	}
 
+	if util.CheckPathExists(EnvFilePath) && !util.CheckPathExists(co.DevfilePath) {
+		log.Warningf("Found a dangling env file without a devfile, overriding it")
+		if err := util.DeletePath(EnvFilePath); err != nil {
+			return err
```

Comment:
Can we replace:
```go
if err := util.DeletePath(EnvFilePath); err != nil {
    return err
```
with
```go
return util.DeletePath(EnvFilePath)
```
Of course, only if it doesn't seem to hurt the readability. I asked this because I found a few occurrences of something like it in odo code.

Reply:
we need to return the error **only if there is one i.e. its not nil**. What you are suggesting is to return irrespective of an error which is logically incorrect.

Reply:
Hmm. Second time in a week that I suggested something like this https://github.com/openshift/odo/pull/4311#discussion_r547119944. Something's wrong in the way I'm looking at these things.